src/lib.rs

```diff
@@ -152,22 +152,8 @@ mod util;
 #[allow(dead_code)]
 mod util_libc;
 
-// std-only trait definitions (also need for use_file)
-#[cfg(any(
-    feature = "std",
-    target_os = "android",
-    target_os = "dragonfly",
-    target_os = "emscripten",
-    target_os = "freebsd",
-    target_os = "haiku",
-    target_os = "illumos",
-    target_os = "linux",
-    target_os = "macos",
-    target_os = "netbsd",
-    target_os = "openbsd",
-    target_os = "redox",
-    target_os = "solaris",
-))]
+// std-only trait definitions
+#[cfg(feature = "std")]
 mod error_impls;
```

Comment:
What happens in this case when the `getrandom` crate is used both by `std` and by other crates wanting `std` features?

Reply:
IIUC right now `getrandom` will be compiled into `std`, so enabling features for it during usual builds will not influence the copy of the crate used by `std`. Although I am not sure how it will work with rust-lang/rfcs#2663.